pygments/lexers/solidity.py

```diff
@@ -59,7 +59,7 @@ class SolidityLexer(RegexLexer):
             (datatype, Keyword.Type),
             include('constants'),
             (r'[a-zA-Z_]\w*', Text),
-            (r'[!<=>+*/-]', Operator),
+            (r'[~!%^&*+=|?:<>/-]', Operator),
```

Comment:
`:` is now recognized as both `Operator` and `Punctuation` - which makes more sense?

Reply:
good catch. it is somewhat ambiguous. `:` can be used in the shorthand if operator `a ? b : c` or also in structs like `{x : y}`. I would stick with Operator.